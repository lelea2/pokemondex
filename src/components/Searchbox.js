import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './styles/Searchbox.css';
import t from '../translations';

class Searchbox extends Component {

  constructor(props) {
    super(props);
    this.state = {
      searchText: ''
    };
  }

  handleSubmit(e) {
    e.preventDefault();
    this.props.searchPokemon(this.state.searchText);
  }

  handleOnChange(e) {
    const searchText = e.target.value;
    this.setState({
      searchText
    });
  }

  resetText() {
    this.setState({
      searchText: ''
    });
  }

  render() {
    return (
      <form className="searchbox" onSubmit={this.handleSubmit.bind(this)}>
        <input
          className="inputSearch"
          placeholder="search"
          onChange={this.handleOnChange.bind(this)}
          value={this.state.searchText}
        />
        <a onClick={this.resetText.bind(this)}><i>{t.resetText}</i></a>
      </form>
    );
  }
}

Searchbox.propTypes = {
  searchPokemon: PropTypes.func
};

export default Searchbox;