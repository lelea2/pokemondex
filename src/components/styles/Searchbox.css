.searchbox {
  width: 90%;
  display: block;
  margin: 10px auto;
}

.searchbox .inputSearch {
  width: 100%;
  padding: 10px 6px;
  border: 1px solid #ddd;
  border-radius: 2px;
  font-size: 20px;
}

.searchbox a {
  cursor: pointer;
  color: #0000ff;
}