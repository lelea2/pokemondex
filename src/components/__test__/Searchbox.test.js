import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Searchbox from '../Searchbox';

describe('Component: Searchbox', () => {
  it("Searchbox renders correctly", () => {
    const test = renderer.create(<Searchbox searchPokemon={() => null} />).toJSON();
    expect(test).toMatchSnapshot();
  });

  it("Searchbox - onchange behavior", () => {
    const event = {
      preventDefault() {},
      target: { value: 'pikachu' }
    };
    const component = shallow(<Searchbox searchPokemon={() => null} />);
    component.find('input').simulate('change', event);
    expect(component.find('input').props().value).toEqual('pikachu');
  });

  it("Searchbox - onchange behavior", () => {
    const searchPokemonMock = jest.fn();
    const event = {
      preventDefault() {},
      target: { value: 'pikachu' }
    };
    const component = shallow(<Searchbox searchPokemon={searchPokemonMock} />);
    component.find('input').simulate('change', event);
    component.find('form').simulate('submit');
    expect(searchPokemonMock).toHaveBeenCalledWith('pikachu');
  });
});
