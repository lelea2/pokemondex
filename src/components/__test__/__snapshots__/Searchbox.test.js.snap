// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Searchbox Searchbox renders correctly 1`] = `
<form
  className="searchbox"
  onSubmit={[Function]}
>
  <input
    className="inputSearch"
    onChange={[Function]}
    placeholder="search"
    value=""
  />
  <a
    onClick={[Function]}
  >
    <i>
      Reset Text
    </i>
  </a>
</form>
`;
